sdk/azidentity/workload_identity.go

```diff
@@ -139,3 +156,239 @@ func (w *WorkloadIdentityCredential) getAssertion(context.Context) (string, erro
 	}
 	return w.assertion, nil
 }
+
+func parseAndValidateCustomTokenEndpoint(endpoint string) (*url.URL, error) {
+	tokenEndpoint, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse custom token endpoint URL %q: %w", endpoint, err)
```

Comment:
Using `%w` means that `errors.Is` and `errors.As` can access the underlying error type. This is usually a mistake. Can you please change this to `%s`.

Reply:
updated!